packages/lab/src/slider/internal/SliderThumb.tsx

```diff
@@ -0,0 +1,82 @@
+import { Label, makePrefixer } from "@salt-ds/core";
+import { clsx } from "clsx";
+import { type ComponentPropsWithoutRef, useState } from "react";
+import type { ActiveThumbIndex, ThumbIndex } from "../types";
+import { useSliderContext } from "./SliderContext";
+import { useKeyDownThumb } from "./useKeyDownThumb";
+import { getPercentage } from "./utils";
+
+const withBaseName = makePrefixer("saltSliderThumb");
+
+export interface SliderThumbProps extends ComponentPropsWithoutRef<"div"> {
+  index: ThumbIndex;
+  activeThumb: ActiveThumbIndex;
+  setActiveThumb: (index: ActiveThumbIndex) => void;
+}
+
+export function SliderThumb(props: SliderThumbProps): JSX.Element {
+  const { index, activeThumb, setActiveThumb } = props;
+
+  const [focussed, setFocussed] = useState(false);
+
+  const { min, max, step, value, onChange, ariaLabel } = useSliderContext();
+
+  const onKeyDown = useKeyDownThumb(min, max, step, value, onChange, index);
+
+  const thumbValue = value[index];
+
+  const thumbPosition = getPercentage(min, max, thumbValue as number);
```

Comment:
We shouldn't cast in these cases, thumbValue could be undefined in it's types which will break